modules/mcc/src/colorspace.hpp

```diff
@@ -0,0 +1,384 @@
+
+private:
+    void setParameter() CV_OVERRIDE;
+};
+
+/** @brief Adobe RGB color space.
+*/
+class AdobeRGB_ : public AdobeRGBBase_
+{
+public:
+    AdobeRGB_(bool linear_ = false)
+        : AdobeRGBBase_(IO::getIOs(D65_2), "AdobeRGB", linear_) {};
+
+private:
+    void setParameter() CV_OVERRIDE;
+};
+
+/** @brief Wide-gamut RGB color space.
+           data from https://en.wikipedia.org/wiki/Wide-gamut_RGB_color_space.
+*/
+class WideGamutRGB_ : public AdobeRGBBase_
+{
+public:
+    WideGamutRGB_(bool linear_ = false)
+        : AdobeRGBBase_(IO::getIOs(D50_2), "WideGamutRGB", linear_) {};
+
+private:
+    void setParameter() CV_OVERRIDE;
+};
+
+/** @brief ProPhoto RGB color space.
+           data from https://en.wikipedia.org/wiki/ProPhoto_RGB_color_space.
+*/
+
+class ProPhotoRGB_ : public AdobeRGBBase_
+{
+public:
+    ProPhotoRGB_(bool linear_ = false)
+        : AdobeRGBBase_(IO::getIOs(D50_2), "ProPhotoRGB", linear_) {};
+
+private:
+    void setParameter() CV_OVERRIDE;
+};
+
+/** @brief DCI-P3 RGB color space.
+           data from https://en.wikipedia.org/wiki/DCI-P3.
+*/
+class DCI_P3_RGB_ : public AdobeRGBBase_
+{
+public:
+    DCI_P3_RGB_(bool linear_ = false)
+        : AdobeRGBBase_(IO::getIOs(D65_2), "DCI_P3_RGB", linear_) {};
+
+private:
+    void setParameter() CV_OVERRIDE;
+};
+
+/** @brief Apple RGB color space.
+           data from http://www.brucelindbloom.com/index.html?WorkingSpaceInfo.html.
+*/
+class AppleRGB_ : public AdobeRGBBase_
+{
+public:
+    AppleRGB_(bool linear_ = false)
+        : AdobeRGBBase_(IO::getIOs(D65_2), "AppleRGB", linear_) {};
+
+private:
+    void setParameter() CV_OVERRIDE;
+};
+
+/** @brief REC_709 RGB color space.
+           data from https://en.wikipedia.org/wiki/Rec._709.
+*/
+class REC_709_RGB_ : public sRGBBase_
+{
+public:
+    REC_709_RGB_(bool linear_)
+        : sRGBBase_(IO::getIOs(D65_2), "REC_709_RGB", linear_) {};
+
+private:
+    void setParameter() CV_OVERRIDE;
+};
+
+/** @brief REC_2020 RGB color space.
+           data from https://en.wikipedia.org/wiki/Rec._2020.
+*/
+class REC_2020_RGB_ : public sRGBBase_
+{
+public:
+    REC_2020_RGB_(bool linear_)
+        : sRGBBase_(IO::getIOs(D65_2), "REC_2020_RGB", linear_) {};
+
+private:
+    void setParameter() CV_OVERRIDE;
+};
+
+/** @brief Enum of the possible types of CAMs.
+*/
+enum CAM
+{
+    IDENTITY,
+    VON_KRIES,
+    BRADFORD
+};
+
+static std::map<std::tuple<IO, IO, CAM>, Mat> cams;
+static const Mat Von_Kries = (Mat_<double>(3, 3) << 0.40024, 0.7076, -0.08081, -0.2263, 1.16532, 0.0457, 0., 0., 0.91822);
+static const Mat Bradford = (Mat_<double>(3, 3) << 0.8951, 0.2664, -0.1614, -0.7502, 1.7135, 0.0367, 0.0389, -0.0685, 1.0296);
+static const std::map<CAM, std::vector<Mat>> MAs = {
+    { IDENTITY, { Mat::eye(Size(3, 3), CV_64FC1), Mat::eye(Size(3, 3), CV_64FC1) } },
+    { VON_KRIES, { Von_Kries, Von_Kries.inv() } },
+    { BRADFORD, { Bradford, Bradford.inv() } }
+};
+
```

Comment:
Reduce initialization of global variables.
Wrap them into functions with "static" variables (functions should return const references).

Reply:
Done